ui/addon/src/main/java/org/jboss/windup/ui/WindupUpdateDistributionCommand.java

```diff
@@ -0,0 +1,157 @@
+import java.util.List;
+
+import javax.inject.Inject;
+
+import net.lingala.zip4j.core.ZipFile;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.jboss.forge.addon.dependencies.Coordinate;
+import org.jboss.forge.addon.dependencies.Dependency;
+import org.jboss.forge.addon.dependencies.DependencyResolver;
+import org.jboss.forge.addon.dependencies.builder.CoordinateBuilder;
+import org.jboss.forge.addon.dependencies.builder.DependencyQueryBuilder;
+import org.jboss.forge.addon.resource.FileResource;
+import org.jboss.forge.addon.ui.command.UICommand;
+import org.jboss.forge.addon.ui.context.UIBuilder;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.context.UIValidationContext;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.windup.util.PathUtil;
+
+/**
+ * Provides a basic UI command updating the rules/migration-core and addons folders with the latest version.
+ *
+ * @author mbriskar
+ */
+public class WindupUpdateDistributionCommand implements UICommand
+{
+
+    @Inject
+    private DependencyResolver dependencyResolver;
+
+    @Override
+    public UICommandMetadata getMetadata(UIContext ctx)
+    {
+        return Metadata.forCommand(getClass()).name("Windup Update Distribution").description("Update the whole windup installation")
+                    .category(Categories.create("Platform", "Migration"));
+    }
+
+    @Override
+    public Result execute(UIExecutionContext context) throws Exception
+    {
+        //TODO:Make this to resolve distribution version (however there is no way to know what is current windup version)
+        if (!RulesetUpdateChecker.rulesetNeedUpdate(dependencyResolver))
+        {
+            return Results.fail("The ruleset is already in the most updated version.");
+        }
+        List<Coordinate> distributionVersions = dependencyResolver.resolveVersions(DependencyQueryBuilder.create(CoordinateBuilder.create()
+                    .setGroupId("org.jboss.windup")
+                    .setArtifactId("windup-distribution")));
+        Coordinate latestDistributionCoordinate = distributionVersions.get(distributionVersions.size() - 1);
+        int i=0;
+        do{
+            i++;
+            latestDistributionCoordinate = distributionVersions.get(distributionVersions.size() - i);
+        }while(latestDistributionCoordinate.isSnapshot()); 
+        
+        Path windupRulesDir = PathUtil.getWindupRulesDir();
+        Path addonsDir = PathUtil.getWindupAddonsDir();
+        Path coreRulesPropertiesPath = windupRulesDir.resolve(RulesetUpdateChecker.RULESET_CORE_DIRECTORY);
```

Comment:
If you feel like it, you could use Forge resource transactions to make sure this all works:

http://forge.jboss.org/addon/org.jboss.forge.addon:resources

```
ResourceTransaction transaction = factory.getTransaction();
try {
   // Starts the transaction
   transaction.begin();

   FileResource<?> resource = factory.create(...);
   // The file won't be updated until commit is performed
   resource.setContents("Hello World");

   String contents = resource.getContents(); // Returns "Hello World"

   FileResource<?> anotherResource = factory.create(...);
   // The file won't be deleted until commit is performed
   anotherResource.delete();

   FileResource<?> newResource = factory.create(...);
   // The file won't be created until commit is performed
   newResource.createNewFile();

   transaction.commit();
} catch (Exception e){
    // Discard all changes since the beginning of this transaction
    transaction.rollback();
}
```